pkgs/applications/video/shaka-packager/default.nix

```diff
@@ -1,62 +1,59 @@
 { lib
 , stdenv
-, fetchurl
+, fetchFromGitHub
 , runCommand
 , shaka-packager
+, cmake
+, ninja
+, python3
+, darwin
+, nix-update-script
 }:
 
-let
-  sources = {
-    "x86_64-linux" = {
-      filename = "packager-linux-x64";
-      hash = "sha256-MoMX6PEtvPmloXJwRpnC2lHlT+tozsV4dmbCqweyyI0=";
-    };
-    aarch64-linux = {
-      filename = "packager-linux-arm64";
-      hash = "sha256-6+7SfnwVRsqFwI7/1F7yqVtkJVIoOFUmhoGU3P6gdQ0=";
-    };
-    x86_64-darwin = {
-      filename = "packager-osx-x64";
-      hash = "sha256-fFBtOp/Zb37LP7TWAEB0yp0xM88cMT9QS59EwW4MrAY=";
-    };
-  };
-
-  source = sources."${stdenv.hostPlatform.system}"
-    or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
-in
 stdenv.mkDerivation (finalAttrs: {
   pname = "shaka-packager";
-  version = "2.6.1";
-
-  src = fetchurl {
-    url = "https://github.com/shaka-project/shaka-packager/releases/download/v${finalAttrs.version}/${source.filename}";
-    inherit (source) hash;
+  version = "3.0.4";
+
+  src = fetchFromGitHub {
+    owner = "shaka-project";
+    repo = finalAttrs.pname;
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-SR5sFcpSbhk1W/epeXMnLfEc/pypuNAieDCzZk4DhEY=";
+    # dependencies are vendored as git submodules
+    fetchSubmodules = true;
```

Comment:
Please unvendor mimalloc, c-ares, protobuf, libxml2, libwebm, libpng, zlib, mbedtls, nlohmann-json, curl, abseil-cpp and googletest.

Reply:
This will take some time and a quite a bit of patching CMakeLists unfortunately, as upstream shaka-packager is set up to produce a mostly static binary (besides glibc).

Reply:
Nonetheless, I'm trying my best!

Reply:
It took me much more time than anticipated, and I ended up running into a wall back then, but today I spent almost the whole day to give it another shot and finally managed to unvendor all dependencies.